iam_role.go

```diff
@@ -114,14 +118,24 @@ func NewIamRoleRequest(options *CreateIamRoleOptions) (*IamRoleRequest, error) {
 		return nil, fmt.Errorf("RoleName option must not be nil")
 	}
 
-	if options.RoleType == nil {
-		return nil, fmt.Errorf("RoleType option must not be nil")
+	trustPolicyExists := options.TrustPolicy != nil
+	roleTypeExists := options.RoleType != nil
+	if trustPolicyExists == roleTypeExists {
+		return nil, fmt.Errorf("Either RoleType or TrustPolicy must be included, but not both")
 	}
 
 	iam := &IamRoleRequest{
 		RoleName: *options.RoleName,
-		RoleType: *options.RoleType,
 	}
+
+	if roleTypeExists {
+		iam.RoleType = *options.RoleType
+	}
```

Comment:
Should we be setting this to nil, as we do with other examples such as template fields.